File: Problems/Algorithms/680. Valid Palindrome II/ValidPalindromeII.kt
class Solution {
    fun validPalindrome(s: String): Boolean {
        var left = 0
        var right = s.length-1
        
        while (left < right) {
            if (s[left] == s[right]) {
                left++
                right--
            } else {
                var sl = StringBuilder(s)
                var sr = StringBuilder(s)
                sl.deleteCharAt(left)
                sr.deleteCharAt(right)
                return sl.reversed().toString().equals(sl.toString()) || sr.reversed().toString().equals(sr.toString())
            }
        }
        
        return true
    }
}

File: Problems/Algorithms/680. Valid Palindrome II/valid_palindrome_II.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        left, right = 0, len(s)-1

        while left < right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:
                sl = s[:left] + s[left+1:]
                sr = s[:right] + s[right+1:]
                return sl == sl[::-1] or sr == sr[::-1]

        return True

